tony-core/src/main/java/com/linkedin/tony/ReusablePort.java

```diff
@@ -107,23 +137,44 @@ private static int getAvailablePort() throws IOException {
    * SO_REUSEPORT.
    * @return the created port
    */
-  static ReusablePort create() throws IOException, InterruptedException {
+  static ReusablePort create() throws IOException {
     ReusablePort reusablePort;
     final int portBindingRetry = 5;
     for (int i = 0; i < portBindingRetry; i++) {
       try {
+        LOG.info("Port binding attempt " + (i + 1) + " ....");
         reusablePort = create(getAvailablePort());
         return reusablePort;
       } catch (BindException ex) {
-        LOG.info("port binding attempt " + (i + 1) + " failed.");
+        LOG.info("Port binding attempt " + (i + 1) + " failed.");
       }
     }
 
     throw new BindException("Unable to bind port after " + portBindingRetry + " attempt(s).");
   }
 
+  private static boolean waitTillPortReserved(int port) {
+    assert RESERVE_PORT_SCRIPT_PATH != null;
```

Comment:
if assert works depending on system setting. Under some setting, assert doesn't work. Is it expected?

Reply:
it's to bypass annoying NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE detected by findbugs  plugin in tony, see https://stackoverflow.com/questions/23746395/getting-rid-of-np-null-on-some-path-from-return-value